### PR TITLE
유저 데이터 입력하기

### PR DESCRIPTION
**Pull request by dicohy27**
ㅇㅇ

### CODE REVIEW

File: src/main/kotlin/fittibackendapp/domain/body/service/UserDetailService.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+class UserDetailService(
+    private val userDetailRepository: UserDetailRepository,
+    private val userRepository: UserRepository,
+    private val userDetailMapStruct: UserDetailMapStruct,
+) {
+    @Transactional
+    fun putUserDetail(
+        putUserDetailInput: PutUserDetailInput,
+        userId: Long,
+    ): UserDetailDto {
```
Comment (uuccu) [inline src/main/kotlin/fittibackendapp/domain/body/service/UserDetailService.kt:25]: 지금 이브랜치에서 DB에 시간대별로 입력해보는거 구현해보자

File: src/main/kotlin/fittibackendapp/dto/UserDetailDto.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    val id: Long,
+    val age: Int?,
+    val height: Double?,
+    val weight: Double?,
+    val muscleMass: Double?,
+    val bodyFat: Double?,
+    val targetWeight: Double?,
+    val targetMuscleMass: Double?,
+    val user: UserDto,
+)
```
Comment (uuccu) [inline src/main/kotlin/fittibackendapp/dto/UserDetailDto.kt:13]: 여기있는 user: UserDto 지우자

File: src/main/resources/db/changelog/changelog-0011.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+databaseChangeLog:
+  - changeSet:
+      id: drop not null contraint for user_detail table
+      author: admin
+      changes:
+        - dropNotNullConstraint:
+            tableName: user_detail
```
Comment (uuccu) [inline src/main/resources/db/changelog/changelog-0011.yml:7]: 여기에 이어서 작성하지 않아도 됨. 
새로 12번 파일 만들어도됨

File: src/main/resources/graphql/domain/body/entity/UserDetail.gql [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    id: ID!,
+    age: Int,
+    height: Float,
+    weight: Float,
+    muscleMass: Float,
+    bodyFat: Float,
+    targetWeight: Float,
+    targetMuscleMass: Float,
+    user: User!,
+}
```
Comment (uuccu) [inline src/main/resources/graphql/domain/body/entity/UserDetail.gql]: 여기서도 User 지우자.

File: src/main/resources/graphql/domain/body/mutation/UserDetailMutation.gql [hunk 5]
```diff
@@ -0,0 +1,4 @@
+extend type Mutation {
+    "UserDetail mutation to put user detail."
+    putUserDetail(putUserDetailInput: PutUserDetailInput!): UserDetail!
+}
```
Comment (uuccu) [inline src/main/resources/graphql/domain/body/mutation/UserDetailMutation.gql]: 주석 설명 제대로 만들자.

**Review by uuccu — CHANGES_REQUESTED** [review]

**Comment by dicohy27** [comment]
ㅇㅋ염

**Comment by uuccu** [comment]
### Code Coverage
|File|Coverage [0.00%]|
|:-|:-:|
|[src/main/kotlin/fittibackendapp/domain/body/entity/UserDetail.kt](https://github.com/FITTI-APP/Fitti-Backend-App/blob/4636bc199dd79b399596972976a8805ddc82304c/src%2Fmain%2Fkotlin%2Ffittibackendapp%2Fdomain%2Fbody%2Fentity%2FUserDetail.kt)|0.00%|
|[src/main/kotlin/fittibackendapp/domain/body/graphql/mutation/UserDetailMutationMapper.kt](https://github.com/FITTI-APP/Fitti-Backend-App/blob/4636bc199dd79b399596972976a8805ddc82304c/src%2Fmain%2Fkotlin%2Ffittibackendapp%2Fdomain%2Fbody%2Fgraphql%2Fmutation%2FUserDetailMutationMapper.kt)|0.00%|
|[src/main/kotlin/fittibackendapp/domain/body/graphql/mutation/input/PutUserDetailInput.kt](https://github.com/FITTI-APP/Fitti-Backend-App/blob/4636bc199dd79b399596972976a8805ddc82304c/src%2Fmain%2Fkotlin%2Ffittibackendapp%2Fdomain%2Fbody%2Fgraphql%2Fmutation%2Finput%2FPutUserDetailInput.kt)|0.00%|
|[src/main/kotlin/fittibackendapp/domain/body/service/UserDetailService.kt](https://github.com/FITTI-APP/Fitti-Backend-App/blob/4636bc199dd79b399596972976a8805ddc82304c/src%2Fmain%2Fkotlin%2Ffittibackendapp%2Fdomain%2Fbody%2Fservice%2FUserDetailService.kt)|0.00%|
|[src/main/kotlin/fittibackendapp/dto/UserDetailDto.kt](https://github.com/FITTI-APP/Fitti-Backend-App/blob/4636bc199dd79b399596972976a8805ddc82304c/src%2Fmain%2Fkotlin%2Ffittibackendapp%2Fdto%2FUserDetailDto.kt)|0.00%|
|[src/main/kotlin/fittibackendapp/exception/NotFoundUserDetailException.kt](https://github.com/FITTI-APP/Fitti-Backend-App/blob/4636bc199dd79b399596972976a8805ddc82304c/src%2Fmain%2Fkotlin%2Ffittibackendapp%2Fexception%2FNotFoundUserDetailException.kt)|0.00%|

|Total Project Coverage|0.00%|:x:|
|:-|:-:|:-:|

**Comment by uuccu** [comment]
테스트코드는 신경쓰지말고 PR리뷰 반영부터하삼

**Comment by dicohy27** [comment]
반영 했잖음

**Review by uuccu — APPROVED** [review]
반영되어있네. commit 버젼 다르게 봤나봄

**Pull request merged**